### PR TITLE
Phemex fetchFundingHistory

### PR DESCRIPTION
**Pull request by Dan-krm**
Added fetchFundingHistory to Phemex:

Set swap to true and the remaining swap related methods to false or undefined.

```
phemex.fetchFundingHistory (BTC/USD:BTC)
2022-05-07T01:06:56.179Z iteration 0 passed in 596 ms

symbol | code |     timestamp |                 datetime | id |     amount
--------------------------------------------------------------------------
BTCUSD |  BTC | 1651881600000 | 2022-05-07T00:00:00.000Z |    | 0.00000005
1 objects
```

### CODE REVIEW

File: js/phemex.js [hunk 1]
```diff
@@ -44,8 +45,15 @@ module.exports = class phemex extends Exchange {
                 'fetchCurrencies': true,
                 'fetchDepositAddress': true,
                 'fetchDeposits': true,
+                'fetchFundingHistory': true,
+                'fetchFundingRate': undefined,
```
Comment (samgermain) [inline js/phemex.js:49]: Just don't add this line, it will be true in your other PR

File: js/phemex.js [hunk 2]
```diff
@@ -44,8 +45,15 @@ module.exports = class phemex extends Exchange {
...
+                'fetchFundingRates': false,
+                'fetchFundingRateHistories': false,
+                'fetchFundingRateHistory': false,
                 'fetchIndexOHLCV': false,
+                'fetchLeverage': false,
                 'fetchLeverageTiers': false,
+                'fetchMarketLeverageTiers': false,
```
Comment (samgermain) [inline js/phemex.js:56]: This information is in the market under `market['info']['riskLimits']`

Comment (samgermain) [reply js/phemex.js:56]: It should be implemented similarly to `ascendex` and `xena`

File: js/phemex.js [hunk 3]
```diff
@@ -24,9 +24,10 @@ module.exports = class phemex extends Exchange {
                 'CORS': undefined,
                 'spot': true,
                 'margin': false,
-                'swap': undefined, // has but not fully implemented
+                'swap': true,
                 'future': false,
                 'option': false,
+                'addMargin': undefined,
```
Comment (samgermain) [inline js/phemex.js:30]: I think that this endpoint might be it https://github.com/phemex/phemex-api-docs/blob/master/Public-Contract-API-en.md#change-position-risklimit

File: js/phemex.js [hunk 4]
```diff
@@ -55,13 +63,17 @@ module.exports = class phemex extends Exchange {
...
+                'fetchPositionsRisk': false,
                 'fetchPremiumIndexOHLCV': false,
                 'fetchTicker': true,
                 'fetchTrades': true,
                 'fetchTradingFee': false,
                 'fetchTradingFees': false,
                 'fetchWithdrawals': true,
+                'reduceMargin': undefined,
                 'setLeverage': true,
+                'setMarginMode': false,
```
Comment (samgermain) [inline js/phemex.js:75]: do you know how to place a cross/isolated order on phemex? I couldn't find anything about it in their api docs

Comment (Dan-krm) [reply js/phemex.js:75]: The docs don't have many details but using the exchange I don't think you're allowed to set the leverage for cross on Phemex it's just a predetermined amount like for btc its 0.83x and if you set the leverage to anything else it seems to make it an isolated position

Comment (Dan-krm) [reply js/phemex.js:75]: Yeah that's how it works https://phemex.com/user-guides/how-do-i-adjust-the-margin-on-my-position

File: js/phemex.js (hunk 3 above)
Comment (Dan-krm) [reply js/phemex.js]: I'll try this out

File: js/phemex.js [hunk 5]
```diff
@@ -2783,6 +2791,65 @@ module.exports = class phemex extends Exchange {
...
+            result.push ({
+                'info': entry,
+                'symbol': this.safeString (entry, 'symbol'),
+                'code': this.safeCurrencyCode (this.safeString (entry, 'currency')),
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+                'id': undefined,
+                'amount': this.fromEv (this.safeString (entry, 'execFeeEv'), market),
+            });
+        }
```
Comment (samgermain) [inline js/phemex.js:2849]: I don't think other exchanges have been doing this, but can we separate this out into a `parseFundingHistory` method?

**Pull request merged**